### PR TITLE
Add FlexIO Implementations for 1010 and 1020

### PR DESCRIPTION
**Pull request by Florob**
Implemented in accordance with the respective reference manual.
i.MX RT1020 was also tested on actual hardware.

### CODE REVIEW

File: src/imxrt1010/flexio.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+flexio!(module: 1, offset: 17, pad: GPIO_SD_11, alt: 4, daisy: None);
+flexio!(module: 1, offset: 18, pad: GPIO_SD_12, alt: 4, daisy: None);
+flexio!(module: 1, offset: 19, pad: GPIO_SD_13, alt: 4, daisy: None);
+flexio!(module: 1, offset: 20, pad: GPIO_AD_00, alt: 4, daisy: None);
+flexio!(module: 1, offset: 21, pad: GPIO_AD_09, alt: 4, daisy: None);
+flexio!(module: 1, offset: 22, pad: GPIO_AD_10, alt: 4, daisy: None);
+flexio!(module: 1, offset: 23, pad: GPIO_AD_11, alt: 4, daisy: None);
+flexio!(module: 1, offset: 24, pad: GPIO_AD_12, alt: 4, daisy: None);
+flexio!(module: 1, offset: 25, pad: GPIO_AD_13, alt: 4, daisy: None);
+flexio!(module: 1, offset: 26, pad: GPIO_AD_14, alt: 4, daisy: None);
```
Comment (mciantyre) [inline src/imxrt1010/flexio.rs:35]: imxrt-ral defines the FlexIO instance with a `0` const generic. These `module: 1` definitions will cause a type mismatch for anyone using type constraints to check module equality.

The recommendation is to use a 0. The 1010 FlexPWM pin implementations take this approach, and it lets us type check against the imxrt-ral 1010 FlexPWM instance.

https://github.com/imxrt-rs/imxrt-iomuxc/blob/03c3f2baac5dd6f09df36d82bffe2d7d954d71a9/src/imxrt1010/flexpwm.rs#L1-L4

Mind if I change this 1 to a 0?

File: src/imxrt1020/flexio.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+flexio!(module: 1, offset: 22, pad: GPIO_EMC_18, alt: 4, daisy: None);
+flexio!(module: 1, offset: 23, pad: GPIO_EMC_19, alt: 4, daisy: None);
+flexio!(module: 1, offset: 24, pad: GPIO_EMC_20, alt: 4, daisy: None);
+flexio!(module: 1, offset: 25, pad: GPIO_EMC_21, alt: 4, daisy: None);
+flexio!(module: 1, offset: 26, pad: GPIO_EMC_22, alt: 4, daisy: None);
+flexio!(module: 1, offset: 27, pad: GPIO_EMC_23, alt: 4, daisy: None);
+flexio!(module: 1, offset: 28, pad: GPIO_EMC_24, alt: 4, daisy: None);
+flexio!(module: 1, offset: 29, pad: GPIO_EMC_25, alt: 4, daisy: None);
+flexio!(module: 1, offset: 30, pad: GPIO_EMC_26, alt: 4, daisy: None);
+flexio!(module: 1, offset: 31, pad: GPIO_EMC_27, alt: 4, daisy: None);
```
Comment (mciantyre) [inline src/imxrt1020/flexio.rs:40]: Same question:

> Mind if I change this 1 to a 0?

I want to make sure this wouldn't affect any of your 1020 testing.

**Review by mciantyre — COMMENTED** [review]
Compared against the reference manuals, and definitions LGTM. Inline comments ask about imxrt-ral compatibility for these chips.

File: src/imxrt1020/flexio.rs (hunk 2 above)
Comment (ggVGc) [reply src/imxrt1020/flexio.rs:40]: This is exactly the issue I stumbled upon when trying to adapt https://github.com/Finomnis/ws2812-flexio to `imxrt1011`, so setting module to 0 seems correct to me.

**Comment by Florob** [comment]
Yes, I believe changing both cases to module `0` is correct. I've not yet worked with any abstractions based on this, beyond utilizing the `OFFSET` value.

**Comment by mciantyre** [comment]
Thanks for putting this together. Released in 0.2.3.

**Pull request merged**